test/qa_settings.cpp

```diff
@@ -102,17 +102,22 @@ struct Source : public node<Source<T>> {
     }
 };
 
+using TestBlockDoc = Doc<R""(
+some test doc documentation
+)"">;
+
 template<typename T>
-struct TestBlock : public node<TestBlock<T>> {
-    IN<T>          in;
-    OUT<T>         out;
-    T              scaling_factor = static_cast<T>(1);
-    std::string    context;
-    std::int32_t   n_samples_max = -1;
-    float          sample_rate   = 1000.0f;
-    std::vector<T> vector_setting{ T(3), T(2), T(1) };
-    int            update_count = 0;
-    bool           debug        = false;
+struct TestBlock : public node<TestBlock<T>, BlockingIO, TestBlockDoc, SupportedTypes<float, double>> {
+    IN<T>  in;
+    OUT<T> out;
+    // parameters
+    A<T, "scaling factor", Visible, Doc<"y = a * x">, Unit<"As">> scaling_factor = static_cast<T>(1); // N.B. unit 'As' = 'Coulomb'
+    A<std::string, "context information", Visible>                context;
+    std::int32_t                                                  n_samples_max = -1;
+    float                                                         sample_rate   = 1000.0f;
+    std::vector<T>                                                vector_setting{ T(3), T(2), T(1) };
+    int                                                           update_count = 0;
+    bool                                                          debug        = false;
```

Comment:
I commented on this because these lines show up in the diff even though only whitespace was changed. This is a huge downside of alignment, which is why I would turn those options off for clang-format. (Whether these lines were mine doesn't matter much. I followed the clang-format rules as defined in the project root. And now those rules start to bite.) For the sake of `git blame` and to reduce clutter in diffs I strongly prefer patches to not change formatting of unrelated lines. If re-formatting is necessary do it in a formatting-only commit.